docs/design/vm_data_protection-design.md

```diff
@@ -0,0 +1,502 @@
+- **Repository Management**: Support for multiple logical backup repositories within a single namespace, associated with a shared BSL.
+
+### Out of Scope
+
+The following are explicitly out of scope for this project phase:
+
+- **Full VM Protection**: This design does not modify or replace OADP’s snapshot-based backup and restore for entire VMs or persistent volumes.
+- **Application Quiescing**: The system will not implement application-level consistency mechanisms (e.g., pausing databases). Ensuring data consistency remains the user's responsibility.
+- **Block-Level Operations**: Backup or restore of raw block devices, unmounted partitions, or disk images is not supported.
+- **Graphical User Interface (UI)**: No integration with the OpenShift Console or other web-based user interfaces is included.
+- **Client Lifecycle Management**: Automated deployment, installation, or upgrading of the in-guest client software is out of scope.
+- **External VM Access**: The design does not introduce new mechanisms for cluster-initiated access to guest VMs (e.g., SSH injection, `virt`-based access).
+- **New Storage Backends**: This effort does not introduce support for new storage backend types. Only those already supported by OADP will be used.
+- **Advanced Observability**: Detailed metrics, dashboards, or alerting features for file-level backup operations are not addressed in this phase.
+
+## Goals
+
+By delivering the features defined in this scope, the VMDP design aims to achieve the following strategic objectives:
+
+- **Empower End Users**: Provide VM users with secure, self-service capabilities to back up and restore their own application data, without needing elevated privileges or administrator intervention.
+- **Enhance Data Protection Granularity**: Address a critical gap between infrastructure-level recovery and application-level data needs by supporting fine-grained, file-specific protection.
+- **Maintain a Secure, Zero-Trust Model**: Ensure that all client-server interactions are strongly authenticated and authorized using OpenShift-native mechanisms, minimizing risk and privilege exposure.
+- **Integrate Seamlessly with OADP**: Leverage existing OADP concepts and APIs (such as `BackupStorageLocation`) to ensure consistent behavior and ease of adoption by administrators.
+- **Deliver a Platform-Agnostic Solution**: Support major operating systems commonly used within guest VMs to maximize utility across enterprise environments.
+
+## High-Level Design
+
+The **Virtual Machine Data Protection (VMDP)** feature introduces a OpenShift-native, client-server architecture that enables file-level backup and restore operations initiated from within OpenShift Virtual Machines. The system integrates seamlessly with OpenShift APIs and the existing OADP infrastructure, while preserving a clear separation of responsibilities between cluster administrators and VM users.
+
+### Architecture Overview
```

Comment:
How do administrators enable VMDP? Should this integrate with DPA configuration or be standalone CRD management? Please clarify this aspect in the design.